server/src/repositories/storage.repository.ts

```diff
@@ -186,7 +186,8 @@ export class StorageRepository implements IStorageRepository {
   }
 
   private asGlob(pathsToCrawl: string[]): string {
-    const base = pathsToCrawl.length === 1 ? pathsToCrawl[0] : `{${pathsToCrawl.join(',')}}`;
+    const escapedPaths = pathsToCrawl.map((path) => escapePath(path));
+    const base = escapedPaths.length === 1 ? escapedPaths[0] : `{${escapedPaths.join(',')}}`;
```

Comment:
This code feels a bit redundant since `Array.join` should work fine on a singleton array. I just left it as is but can tweak.